### PR TITLE
ARROW-11126: [Rust] Document and test ARROW-10656

### PR DESCRIPTION
**Pull request by nevi-me**
I hadn't pushed these 2 commits to the ARROW-10656 branch

### CODE REVIEW

File: rust/arrow/src/record_batch.rs [hunk 1]
```diff
@@ -80,6 +80,10 @@ impl RecordBatch {
         Ok(RecordBatch { schema, columns })
     }
 
+    /// Creates a [RecordBatch` from a schema and columns, with additional options,
```
Comment (alamb) [inline rust/arrow/src/record_batch.rs:83]:
```suggestion
    /// Creates a [`RecordBatch`] from a schema and columns, with additional options,
```

**Review by alamb — APPROVED** [review]
Thanks @nevi-me  -- looks good to me. 

Clippy seems to be unhappy with single name character variable names: https://github.com/apache/arrow/runs/1645638518?check_suite_focus=true

```
error: 7 bindings with single-character names in scope
    --> arrow/src/datatypes.rs:1913:13
     |
1913 |         let a = DataType::List(Box::new(Field::new("item", DataType::Int32, true)));
     |             ^
1914 |         let b = DataType::List(Box::new(Field::new("array", DataType::Int32, true)));
     |             ^
1915 |         let c = DataType::List(Box::new(Field::new("item", DataType::Int32, false)));
     |             ^
1916 |         let d = DataType::List(Box::new(Field::new("item", DataType::UInt32, true)));
     |             ^
...
1922 |         let e =
     |             ^
1923 |             DataType::FixedSizeList(Box::new(Field::new("item", a.clone(), false)), 3);
1924 |         let f =
     |             ^
1925 |             DataType::FixedSizeList(Box::new(Field::new("array", b.clone(), false)), 3);
1926 |         let g = DataType::FixedSizeList(
     |             ^
...
```

File: rust/arrow/src/record_batch.rs (hunk 1 above)
Comment (nevi-me) [reply rust/arrow/src/record_batch.rs]: Thanks, I meant for it to be `RecordBatch`, as the link has little value. I think I removed the wrong character while changing it

**Review by jorgecarleitao — APPROVED** [review]
LGTM. Thanks!

**Pull request closed without merging**